api/v1alpha1/jobset_webhook.go

```diff
@@ -56,6 +57,13 @@ var _ webhook.Validator = &JobSet{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *JobSet) ValidateCreate() error {
+	for i := range r.Spec.ReplicatedJobs {
+		if r.Spec.ReplicatedJobs[i].Network == nil || !*r.Spec.ReplicatedJobs[i].Network.EnableDNSHostnames ||
```

Comment:
do it like this:

```
enableDNSHostnames := .Spec.ReplicatedJobs[i].Network != nil &&
Spec.ReplicatedJobs[i].Network.EnableDNSHostnames != nil &&  *r.Spec.ReplicatedJobs[i].Network.EnableDNSHostnames
indexed := r.Spec.ReplicatedJobs[i].Template.Spec.CompletionMode != nil &&
				*r.Spec.ReplicatedJobs[i].Template.Spec.CompletionMode == batchv1.IndexedCompletion

if enableDNSHostnames && !indexed {
  return fmt.Errorf("EnableDNSHostnames requires job to be in Indexed completion mode")
}

```

Reply:
done fot it. In fact, I want to assign a new var.
like this ``replicatedJobs := r.Spec.ReplicatedJobs`` and the rest of ``r.Spec.ReplicatedJobs[i]`` can change to ``replicatedJobs[i]``. what do you think?